test/framework/nutanix.go

```diff
@@ -119,233 +125,273 @@ func RequiredNutanixEnvVars() []string {
+// Ubuntu126Template returns NutanixFiller by reading the env var and setting machine config's
 // image name parameter in the spec.
-func UpdateNutanixUbuntuTemplate126Var() api.NutanixFiller {
-	return api.WithNutanixStringFromEnvVar(nutanixTemplateNameUbuntu126Var, api.WithNutanixMachineTemplateImageName)
+func (n *Nutanix) Ubuntu126Template(osVersion ...string) api.NutanixFiller {
+	return n.templateForKubeVersionAndOS(anywherev1.Ubuntu, anywherev1.Kube126, osVersion...)
 }
 
-// UpdateNutanixUbuntuTemplate127Var returns NutanixFiller by reading the env var and setting machine config's
+// Ubuntu127Template returns NutanixFiller by reading the env var and setting machine config's
 // image name parameter in the spec.
-func UpdateNutanixUbuntuTemplate127Var() api.NutanixFiller {
-	return api.WithNutanixStringFromEnvVar(nutanixTemplateNameUbuntu127Var, api.WithNutanixMachineTemplateImageName)
+func (n *Nutanix) Ubuntu127Template(osVersion ...string) api.NutanixFiller {
+	return n.templateForKubeVersionAndOS(anywherev1.Ubuntu, anywherev1.Kube127, osVersion...)
 }
 
 // ClusterStateValidations returns a list of provider specific ClusterStateValidations.
-func (s *Nutanix) ClusterStateValidations() []clusterf.StateValidation {
+func (n *Nutanix) ClusterStateValidations() []clusterf.StateValidation {
 	return []clusterf.StateValidation{}
 }
+
+func (n *Nutanix) getDevRelease() *releasev1.EksARelease {
+	n.t.Helper()
+	if n.devRelease == nil {
+		latestRelease, err := getLatestDevRelease()
+		if err != nil {
+			n.t.Fatal(err)
+		}
+		n.devRelease = latestRelease
+	}
+
+	return n.devRelease
+}
+
+func (n *Nutanix) templateForDevRelease(osFamily anywherev1.OSFamily, kubeVersion anywherev1.KubernetesVersion, osVersion ...string) string {
+	n.t.Helper()
+	return n.templatesRegistry.templateForRelease(n.t, osFamily, n.getDevRelease(), kubeVersion, osVersion...)
+}
+
+// envVarForTemplate looks for explicit configuration through an env var: "T_NUTANIX_TEMPLATE_{osFamily}_{eks-d version}"
+// eg: T_NUTANIX_TEMPLATE_UBUNTU_KUBERNETES_1_23_EKS_22.
+func (n *Nutanix) envVarForTemplate(osFamily, osVersion, eksDName string) string {
+	templateEnvVar := fmt.Sprintf("T_NUTANIX_TEMPLATE_NAME_%s_%s", strings.ToUpper(osFamily), strings.ToUpper(strings.ReplaceAll(eksDName, "-", "_")))
+	if osVersion != defaultOSVersions[osFamily] {
+		templateEnvVar = fmt.Sprintf("T_NUTANIX_TEMPLATE_NAME_%s_%s_%s", strings.ToUpper(osFamily), osVersion, strings.ToUpper(strings.ReplaceAll(eksDName, "-", "_")))
+	}
+	return templateEnvVar
+}
+
+// defaultNameForTemplate looks for a template: "{eks-d version}-{osFamily}"
+// eg: kubernetes-1-23-eks-22-ubuntu.
+func (n *Nutanix) defaultNameForTemplate(osFamily, osVersion, eksDName string) string {
+	defaultTemplateName := fmt.Sprintf("%s-%s", strings.ToLower(eksDName), strings.ToLower(osFamily))
+	if osVersion != defaultOSVersions[osFamily] {
```

Comment:
could we add a comment to all these methods explaining why we do this? (I'm assuming it's for backwards compat?)
Although we could just change all the current env vars to follow this new pattern and the code will be simpler.

Reply:
Will do

Reply:
Done